src/test/java/io/supertokens/test/accountlinking/GetUserByAccountInfoTest.java

```diff
@@ -0,0 +1,366 @@
+        assertEquals(0, AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, false, "test3@gmail.com", null, "google", "userid1").length);
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testListUsersByAccountInfoForUnlinkedAccountsWithUnionOption() throws Exception {
+        String[] args = {"../"};
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args, false);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{
+                        EE_FEATURES.ACCOUNT_LINKING, EE_FEATURES.MULTI_TENANCY});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        AuthRecipeUserInfo user1 = createEmailPasswordUser(process.getProcess(), "test1@example.com", "password1");
+        AuthRecipeUserInfo user2 = createThirdPartyUser(process.getProcess(), "google", "userid1", "test2@example.com");
+        AuthRecipeUserInfo user3 = createPasswordlessUserWithEmail(process.getProcess(), "test3@example.com");
+        AuthRecipeUserInfo user4 = createPasswordlessUserWithPhone(process.getProcess(), "+919876543210");
+
+        TenantIdentifierWithStorage tenantIdentifierWithStorage = TenantIdentifier.BASE_TENANT.withStorage(StorageLayer.getBaseStorage(process.getProcess()));
+        {
+            AuthRecipeUserInfo[] users = AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, true, "test1@example.com", "+919876543210", null, null);
+            assertEquals(2, users.length);
+            assertTrue(Arrays.asList(users).contains(user1));
+            assertTrue(Arrays.asList(users).contains(user4));
+        }
+        {
+            AuthRecipeUserInfo[] users = AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, true, "test1@example.com", null, "google", "userid1");
+            assertEquals(2, users.length);
+            assertTrue(Arrays.asList(users).contains(user1));
+            assertTrue(Arrays.asList(users).contains(user2));
+        }
+        {
+            AuthRecipeUserInfo[] users = AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, true, null, "+919876543210", "google", "userid1");
+            assertEquals(2, users.length);
+            assertTrue(Arrays.asList(users).contains(user4));
+            assertTrue(Arrays.asList(users).contains(user2));
+        }
+        {
+            AuthRecipeUserInfo[] users = AuthRecipe.getUsersByAccountInfo(tenantIdentifierWithStorage, true, "test1@example.com", "+919876543210", "google", "userid1");
+            assertEquals(3, users.length);
+            assertTrue(Arrays.asList(users).contains(user1));
+            assertTrue(Arrays.asList(users).contains(user2));
+            assertTrue(Arrays.asList(users).contains(user4));
+        }
+
```

Comment:
need to test with false option too?

Reply:
this test is about union set to true. there is a different test that tests with false value